### ISSUE TITLE
[0.1.0-canary.14+] Utils with object parameters are no longer applied

### ISSUE THREAD

**Issue by arempe93** [modulz/stitches#466]
# Bug report

## Describe the bug

If a util takes in an object as a parameter

```ts
{
  simpleGradient: (config) => ({ from, to, deg = 90 }) => ({
    backgroundImage: `linear-gradient(${deg}deg, ${from} 0%, ${to} 100%)`
  })
}
```

it will not be applied. utils that take in strings, numbers, etc will be applied.

## To Reproduce

https://codesandbox.io/s/stitches-util-object-bug-9yb24

> Switch to canary.12 to see working behavior

## Expected behavior

Utils should be able to define objects as input.

## System information

- Version of Stitches: 0.1.0-canary.14

**Comment by jonathantneal** [comment]
I’ve been waiting for someone to ask about this. 😄

As far as I know, the only way to safely support this would be to force all selectors to include `&`. Otherwise, how would stitches know the difference between `simpleGradient` the property and `simpleGradient` the descendant selector?

**Comment by arempe93** [comment]
right, I don't think there is an elegant way to differentiate completely. I'd propose simply giving utils higher priority.

so if we have a situation like this

```ts
// config utils
{
  article: (config) => ({ color = '$textDark', size }) => ({
    color: color,
    padding: size,

    fontSize: '$lg',
    lineHeight: 2
  })
}

// styled({ ... })
{
  width: '100%',

  article: { size: '$3' }
}
```

we use the util because we know `'article' in utils == true`, as opposed to creating a `.sxabc article` selector.

then you have two escape hatches as a developer:

1. rename the utility (eg: add prefix)
2. use an `'& article':` key just in this instance for more specificity of your intention

**Comment by jonathantneal** [comment]
Let’s add the option I mentioned to the mix, even if it may (or may not) be unpopular:

### Requiring selectors use the nesting selector

```js
// config utils
{
  article: (config) => ({ color = '$textDark', size }) => ({
    color: color,
    padding: size,

    fontSize: '$lg',
    lineHeight: 2
  })
}

// styled({ ... })
{
  width: '100%',

  // pass `{ size: '$3' }` to the `article` utility, otherwise to the `article` property
  article: { size: '$3' }

  // create a nested rule from `& article`
  "& article": { size: '$3' }
}
```

**Comment by dylanklohr** [comment]
Is the `@bp1` prefix that is/was being considered for responsive breakpoints is still on the table, this might be another natural extension of the `@` to signify a utility? 🤔

**Comment by arempe93** [comment]
closing as fixed via #477 -- thanks @jonathantneal
